hal/tests/TESTS/mbed_hal/flash/functional_tests/CMakeLists.txt

```diff
@@ -3,11 +3,16 @@
 
 cmake_minimum_required(VERSION 3.19.0 FATAL_ERROR)
 
-set(MBED_PATH ${CMAKE_CURRENT_SOURCE_DIR}/../../../../.. CACHE INTERNAL "")
+set(MBED_PATH ${CMAKE_CURRENT_SOURCE_DIR}/../../../../../.. CACHE INTERNAL "")
```

Comment:
Good catch, so you actually validated them! 👍

Reply:
Could you move this into its own commit please? It's not really related to making `TEST_SOURCES` mandatory.